### PR TITLE
Fix operations for u16/u32/u64 on Torchx

### PR DESCRIPTION
**Pull request by josevalim**

### CODE REVIEW

**Comment by josevalim** [comment]
Thanks to @polvalente on the pairing!

File: nx/lib/nx/random.ex [hunk 1]
```diff
@@ -129,33 +129,6 @@ defmodule Nx.Random do
...
-
-      32 ->
-        threefry2x32(key, Nx.iota(shape, type: {:s, 64}))
-        |> Nx.as_type({:u, 32})
-
-      _ ->
-        threefry2x32(key, Nx.iota(shape, type: {:s, 64}))
-        |> Nx.as_type({:u, bit_width})
-    end
-  end
```
Comment (josevalim) [inline nx/lib/nx/random.ex:157]: No changes here, I just moved it down.

Comment (josevalim) [reply nx/lib/nx/random.ex:157]: The rest of the changes in the file is to require fewer `as_type`. No changes to the algo.

File: torchx/lib/torchx/backend.ex [hunk 2]
```diff
@@ -808,17 +814,46 @@ defmodule Torchx.Backend do
     raise ArithmeticError, "Torchx does not support complex values for atan2"
   end
 
+  defp bitmask(tensor, {:u, 16}),
+    do: Torchx.bitwise_and(tensor, Torchx.scalar_tensor(0xFFFF, :int, :cpu))
+
+  defp bitmask(tensor, {:u, 32}),
+    do: Torchx.bitwise_and(tensor, Torchx.scalar_tensor(0xFFFF_FFFF, :long, :cpu))
+
+  defp bitmask(tensor, _),
```
Comment (polvalente) [inline torchx/lib/torchx/backend.ex:823]: What happens if we mask with 0xffff_ffff here instead?

Comment (polvalente) [reply torchx/lib/torchx/backend.ex:823]: Or is the idea to emulate as best as possible u64?

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**